### PR TITLE
12.0 mig stock_reserve

### PR DESCRIPTION
**Pull request by hzh0292**
The module stock_reserve_sale depends on this module.

### CODE REVIEW

File: stock_reserve/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+##############################################################################
```
Comment (rousseldenis) [inline stock_reserve/__init__.py]: You can remove this comment

File: stock_reserve/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from . import model
```
Comment (rousseldenis) [inline stock_reserve/__init__.py]: Change to 'models' directory

File: stock_reserve/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+##############################################################################
```
Comment (rousseldenis) [inline stock_reserve/__manifest__.py]: Replace by shortened comments like in https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: stock_reserve/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+{'name': 'Stock Reservation',
+ 'summary': 'Stock reservations on products',
+ 'version': '12.0.1.0.0',
+ 'author': "Camptocamp,Odoo Community Association (OCA)",
+ 'category': 'Warehouse',
+ 'license': 'AGPL-3',
+ 'complexity': 'normal',
+ 'images': [],
```
Comment (rousseldenis) [inline stock_reserve/__manifest__.py]: Remove void entries

File: stock_reserve/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ 'website': "http://www.camptocamp.com",
```
Comment (rousseldenis) [inline stock_reserve/__manifest__.py]: Change following https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: stock_reserve/model/__init__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+##############################################################################
```
Comment (rousseldenis) [inline stock_reserve/model/__init__.py]: Remove this comment

File: stock_reserve/model/product.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
+##############################################################################
```
Comment (rousseldenis) [inline stock_reserve/model/product.py]: Remove this comment

File: stock_reserve/model/stock_reserve.py [hunk 8]
```diff
@@ -0,0 +1,256 @@
+##############################################################################
```
Comment (rousseldenis) [inline stock_reserve/model/stock_reserve.py]: Remove this comment

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some comments.

Please don't remove translation files

**Comment by rven** [comment]
The yaml test file needs to be converted to python:
https://github.com/odoo/odoo/commit/cf853a785ba4b533ddb2e8a717c9ffa6429e7b81

File: stock_reserve/models/stock_reserve.py [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+        'Reservation Move',
+        required=True,
+        readonly=True,
+        ondelete='cascade',
+        index=True)
+    date_validity = fields.Date('Validity Date')
+
+    @api.model
+    def default_get(self, fields_list):
+        """ Fix default values
```
Comment (rousseldenis) [inline stock_reserve/models/stock_reserve.py:43]: IMHO this is quite weird to do such kind of things in default_get for a Model.

In which case default  field attribute does not work ?

File: stock_reserve/models/stock_reserve.py [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+        move = self.move_id or self.env['stock.move'].new(
+            {'product_id': self.product_id})
+        move.onchange_product_id()
+        self.name = move.name
+        self.product_uom = move.product_uom
+
+    @api.onchange('product_uom_qty')
+    def _onchange_quantity(self):
+        """ On change of product quantity avoid negative quantities """
+        if not self.product_id or self.product_uom_qty <= 0.0:
```
Comment (rousseldenis) [inline stock_reserve/models/stock_reserve.py]: You have to use float_compare.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some changes here

File: stock_reserve/models/stock_reserve.py (hunk 9 above)
Comment (hzh0292) [reply stock_reserve/models/stock_reserve.py:43]: Sorry, that's the old code of version 8.0.So I don't know much detail about the logic case.

File: stock_reserve/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+ 'website': "https://github.com/OCA/stock-logistics-warehouse",
+ 'depends': ['stock',
+             ],
+ 'data': ['views/stock_reserve.xml',
+          'views/product.xml',
+          'data/stock_data.xml',
+          'security/ir.model.access.csv',
+          ],
+ 'auto_install': False,
+ 'test': ['test/stock_reserve.yml',
```
Comment (rousseldenis) [inline stock_reserve/__manifest__.py:17]: yml tests are deprecated. Could you convert to xml ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]

File: stock_reserve/__manifest__.py (hunk 11 above)
Comment (hzh0292) [reply stock_reserve/__manifest__.py:17]: Yes,it's a problem.Is there somebody can help me to convert the yml to python?

**Comment by ventor-dev** [comment]
@hzh0292 , any progress on this migration?

**Comment by rousseldenis** [comment]
@hzh0292

**Pull request closed without merging**